starters/adapters/bun/tsconfig.json

```diff
@@ -0,0 +1,3 @@
+{
+  "compilerOptions": { "types": ["bun-types"] }
```

Comment:
To prevent the extra dependency we can define all the `"types"` (dafault + bun ) in the bun tsconfig.json
This revert some of your changes as well

Reply:
Or types can be moved into an `.d.ts` file instead, like in Nuxt or Remix and then referenced in `includes` field. This way typescript will be able to pick types from `$PWD/node_modules/@types`. Also, this way TypeScript should pick bun types from `@types/bun` package without any additional configuration

See the documentation: https://www.typescriptlang.org/tsconfig#types

If `types` field specified, then TS won't pick anything that is not included in the list:

> This tsconfig.json file will only include ./node_modules/@types/node, ./node_modules/@types/jest and ./node_modules/@types/express. Other packages under node_modules/@types/* will not be included.

So suggestion is to create a file, let's call it `qwik.env.d.ts` for example, where we reference `vite/client` types, and then add it to the `includes`.

The template would look like this:

In `qwik.env.d.ts`:

```ts
/// <reference types="vite/client" />
```

Then in `tsconfig.json`:

```json5
{
  "include": ["qwik.env.d.ts", "**/*.ts", "**/*.tsx"]
}
```

Then for those who want to keep using `bun-types` (as they might be updated more frequently, aligned with the runtime releases) instead of `@types/bun`, they can just remove `@types/bun` after they installed the adapter and reference bun-types same way in `qwik.env.d.ts`